packages/tscc/src/external_module_support.ts

```diff
@@ -3,11 +3,11 @@
  * `const localName = global_name_for_the_external_module`.
  * Also transforms `import tslib_any from 'tslib'` to `goog.require("tslib")`.
  */
-import ITsccSpecWithTS from './spec/ITsccSpecWithTS';
```

Comment:
As I understand, you mostly removed references to `ITsccSpecWithTS` out from the codebase and chose to directly reference `TsccSpecWithTS`. I understand that this convention is something could be a subject of debates, was it purely your coding convention? Or, were there any necessity in doing so in order to enable `--strict` compilation?

Reply:
I just did whatever it took to get `--strict` to stop complaining, it frequently remarked that there were mismatches in the types, that we were providing `TsccSpecWithTS` when it was expecting a `ITsccSpecWithTS`. I am not very experienced with TS, this is kind of my foray into it tbh.

I got the impression that the `TsccSpecWithTS` class was basically an updated version of the `ITsccSpecWithTS` class.

Reply:
`ITsccSpecWithTS` provides a tighter type declaration then `TsccSpecWithTS` most of time. For example, when I want to return an immutable array from a method, I do not bother with Array#slice in the method implementation and just modify the corresponding return type annotation in `ITsccSpecWithTS` with `ReadonlyArray`.

Reply:
Technically I did not adjust any of this logic, I only adjusted the types so that they accurately represented what was already happening. I am not in touch enough with the underlying logic to help out much, but I hope to gain familiarity over the coming weeks. In essence, `ITsccSpecWithTS` was only imported here to be incorrectly used as a type definition for an object that was actually `TsccSpecWithTS`.

Reply:
I wouldn't say that it is "incorrect", it compiles without type error in the current setup. If there is any type errors caused with `--strict`, you can adjust the type declaration of `ITsccSpecWithTS` or modify consumer implementation, for example by adding `!` appropriately.

Reply:
You are totally welcome to edit this PR, I just made the changes which silenced the compiler to get a better handle on TS and was not fully cognizant of what was going on with the calls to the various specs.